File: C#/1140 - Flores Florescem da França.cs
/// author Josélio de S. C. Júnior <joseliojrx25@gmail.com>
/// copyright Josélio de S. C. Júnior 2022

using System;
using System.Linq;

class URI {
  static void Main(string[] args) { 
    while (true) {
      string r = Console.ReadLine() ?? "";
  
      if (r == "*") break;
  
      string[] t = r.Split(' ')
        .Select(e => e[0].ToString().ToLower())
        .ToArray();
  
      if (t.All(e => e == t[0])) {
        Console.WriteLine("Y");
      } else Console.WriteLine("N");
    }
  }
}

File: Python_3.9/1140 - Flores Florescem da França.py
# author: Josélio Júnior <joseliojrx25@gmail.com>
# copyright: Josélio Júnior 2022

while True:
  a = input()
  if a == '*':
    break
  a = a.lower().split(' ')
  a = list(map(lambda x: x[0], a))
  b = list(filter(lambda x: x == a[0], a))
  a = 'Y' if len(a) == len(b) else 'N'
  print(a)
